src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -409,36 +412,49 @@ public List<Course> getAllCoursesToRegister() {
      *
      * @param courseId the courseId for which exercises, lectures, exams and learning goals should be fetched
      * @param refresh  if true, this request was initiated by the user clicking on a refresh button
-     * @return a course with all exercises, lectures, exams, learning goals, etc. visible to the user
+     * @return a course with all exercises, lectures, exams, learning goals, etc. visible to the user. In addition, scores per exercise type for each exercise are sent back as an
+     *         optimization.
      */
     // TODO: we should rename this into courses/{courseId}/details
     @GetMapping("courses/{courseId}/for-dashboard")
     @PreAuthorize("hasRole('USER')")
-    public Course getCourseForDashboard(@PathVariable long courseId, @RequestParam(defaultValue = "false") boolean refresh) {
+    public CourseForDashboardDTO getCourseForDashboard(@PathVariable long courseId, @RequestParam(defaultValue = "false") boolean refresh) {
         long timeNanoStart = System.nanoTime();
         log.debug("REST request to get one course {} with exams, lectures, exercises, participations, submissions and results, etc.", courseId);
         User user = userRepository.getUserWithGroupsAndAuthorities();
         Course course = courseService.findOneWithExercisesAndLecturesAndExamsAndLearningGoalsAndTutorialGroupsForUser(courseId, user, refresh);
-        courseService.fetchParticipationsWithSubmissionsAndResultsForCourses(List.of(course), user);
-        logDuration(List.of(course), user, timeNanoStart);
-        return course;
+        List<Course> courses = List.of(course);
+        courseService.fetchParticipationsWithSubmissionsAndResultsForCourses(courses, user);
+        courseService.fetchPlagiarismCasesForCourses(courses, user.getId());
+        CourseForDashboardDTO courseForDashboardDTO = courseScoreCalculationService.getScoresAndParticipationResults(course, user.getId());
+        logDuration(courses, user, timeNanoStart);
```

Comment:
I cannot just add a method that fetches plagiarism cases for one course because I want to do one database call that gets the plagiarism cases for all the exercises in all the courses instead of one database call per course. I was able to refactor the method to directly receive the exercises instead of the courses. This way I don't have to wrap the course in `List.of()` at least for this method. For `fetchParticipationsWithSubmissionsAndResultsForCourses` this is not easily doable because it does require the courses to do a check on them instead of the individual exercise. Again, this method only does *one* database call to get the participations for all the exercises in all the courses. It's also out of scope for this PR so I'm leaving it as it is.

Reply:
I don't agree with your assessment that it creates issues with database calls. It would be only one call in my case too.

However, I agree that it's out of scope.